src/math-history.c

```diff
@@ -94,12 +97,33 @@ math_history_clear(MathHistory *history)
     GList *children, *iter;
 
     history->priv->rows = 0;
+    history->priv->current = 0;
     children = gtk_container_get_children(GTK_CONTAINER(history->priv->listbox));
     for (iter = children; iter != NULL; iter = g_list_next(iter))
           gtk_widget_destroy(GTK_WIDGET(iter->data));
     g_list_free(children);
 }
 
+gpointer
+math_history_get_entry_at(MathHistory *history, int index)
+{
+    if (index >= 0 && index < history->priv->rows)
+        return MATH_HISTORY_ENTRY(gtk_list_box_get_row_at_index(GTK_LIST_BOX(history->priv->listbox), index));
+    return NULL;
+}
+
+void
+math_history_set_current(MathHistory *history, int value)
+{
+    history->priv->current = fmax(0, fmin(history->priv->rows - 1, history->priv->current + value));
```

Comment:
Why did you use fmax/fmin for integer values?

Reply:
`history->priv->current = CLAMP (history->priv->current + value, 0, history->priv->rows - 1);`
https://developer.gnome.org/glib/stable/glib-Standard-Macros.html#CLAMP:CAPS

Reply:
because I didn't know about this Macro :)